russh-keys/src/key.rs

```diff
@@ -269,6 +310,16 @@ impl PublicKey {
                 };
                 verify().unwrap_or(false)
             }
+            #[cfg(not(feature = "openssl"))]
+            PublicKey::RSA { ref key, ref hash } => {
+                let pss = match hash {
+                    SignatureHash::SHA1 => Pss::new::<Sha1>(),
+                    SignatureHash::SHA2_256 => Pss::new::<Sha256>(),
+                    SignatureHash::SHA2_512 => Pss::new::<Sha512>(),
```

Comment:
I didn't debug but those tests are about signing/verifying. I think this should use PKCS#1 v1.5 padding scheme, instead of PSS. See `rsa::pkcs1v15`.

Reply:
ya,
page 15 https://www.rfc-editor.org/rfc/rfc4253
and
page 4 https://datatracker.ietf.org/doc/html/rfc8332

Reply:
that would be `Pkcs1v15Sign`, then?

Reply:
Thanks for the pointers, I have changed this part and `rsa_signature`, but the same tests are failling
```
failures:
    test::test_agent
    test::test_loewenheim
```

Reply:
`buffer` needs to be hashed before passing into `verify`. I guess you could use `VerifyingKey`, that can hash it for you.